exporter/awsemfexporter/grouped_metric.go

```diff
@@ -83,6 +97,130 @@ func addToGroupedMetric(pmd *pdata.Metric, groupedMetrics map[interface{}]*Group
 			}
 		}
 	}
+
+	return nil
+}
+
+type kubernetesObj struct{
+	Container_name string `json:`
+	Docker internalDockerObj `json:`
+	Host string `json:`
+	Labels internalLabelsObj `json:`
+	Namespace_name string `json:`
+	Pod_id string `json:`
+	Pod_name string `json:`
+	Pod_owners internalPodOwnersObj `json:`
+	Service_name string `json:`
+}
+
+type internalDockerObj struct{
+	Container_id string `json:`
+}
+
+type internalLabelsObj struct{
+	App string `json:`
+	Pod_template_hash string `json: "pod-template-hash"`
+}
+
+type internalPodOwnersObj struct{
+	Owner_kind string `json:`
+	Owner_name string `json:`
+}
+
+
+func addKubernetesWrapper(labels map[string]string) error{
+	//create schema
+	schema := kubernetesObj{}
+	schema.Container_name = "container_name"
+	schema.Docker =
+	internalDockerObj{
+		Container_id: "container_id",
+	}
+	schema.Host = "host_name"
+	schema.Labels =
+	internalLabelsObj{
+		App: "app",
+		Pod_template_hash: "pod-template-hash",
+	}
+	schema.Namespace_name = "namespace_name"
+	schema.Pod_id = "pod_id" 
+	schema.Pod_name = "pod_name"
+	schema.Pod_owners = 
+	internalPodOwnersObj{
+			 Owner_kind: "owner_kind" , 
+			 Owner_name: "owner_name" ,
+		}
+	schema.Service_name = "service_name"
+
+	var err error
+	labels["kubernetes"], err = recursivelyFillInStruct(labels,schema)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+
+func recursivelyFillInStruct(labels map[string]string, schema interface{}) (string, error){
+	jsonBytes, err := json.Marshal(schema)
```

Comment:
I haven't looked into deeply how to improve it, but marshalling and unmarshalling and remarshalling seems like too much marshalling that you'd expect in code like this, can it be improved?

Reply:
This can be improved fairly easily by just submitted the schema as a map in the first place which would be a fairly simple change. Alternatively I could convert from a struct to a map using reflects but those are a lot less safe (there are also some external packages that wrap this functionality in a single command but introducing more dependencies also seemed like a poor idea)

The reason I set up the code to accept schemas as structs was I figured that would be more future proof but I am very new to the community to if that isn't the case I'm happy to switch the way schemas are submitted.

As for using a struct directly as a schema this is super messy for type related reasons and is a bad idea in general given the way Go handles reflection.

Reply:
This code is messy. It looks your `schema/kubernetesObj` has all hardcoded value, why do you need do the recursively conversion?

Reply:
The reason to have the code be written in a generic way is so that it will be easy to change the names of incoming metrics or the structure of the kubernetes object in the future. I had considered at some point having the schema be given as input in the config but it was eventually decided against. Is there a specific aspect of the code you find messy or is it just the recursion over all.

Reply:
@mxiamxia @anuraaga @alolita @Aneurysm9 I believe I've dealt with all of the quick issues, if my answer to the above question was unsatisfying or you have any other concerns please let me know - otherwise hopefully we can get this PR merged soon

Reply:
Isn't the schema static? Why do we need a schema instead of just writing it as standard data translation code? This seems to be on the hot path so if we can accomplish the same with more direct code it's better.

Reply:
@anuraaga we made it non-static to make it easy to make changes in the future and to have it do more kinds of reformatting in the future. Because we are connecting two services we don't control we wanted to have some flexibility

Reply:
> Isn't the schema static? Why do we need a schema instead of just writing it as standard data translation code? This seems to be on the hot path so if we can accomplish the same with more direct code it's better.

Agreed.  This feels like it should be a matter of filling in the appropriate labels.  If we find at a later date that a general purpose label translation/transformation framework is required here then we can build it at that point.

Reply:
@Aneurysm9  My main objections to the full hardcoding approach is  1) that in order to distinguish between a lack of a key and an empty string you would need a large number of if statements in a row creating unnecessary code complexity and making things harder to maintain 2) that it make it harder to make changes in the future if there are some slight changes in the format (as mentioned above).

Additionally, given that this approach has the same or lower maintenance costs than the hardcoded solution reverting to a purely hardcoded solution seems to be only a loss.